desc/utils.py

```diff
@@ -487,3 +488,39 @@ def parse_argname_change(arg, kwargs, oldname, newname):
         )
         arg = kwargs.pop(oldname)
     return arg
+
+
+def setdefault(val, default, cond=None):
+    """Return default if val is None or cond is True, otherwise val."""
+    if cond is None:
+        cond = val is not None
+    if cond:
+        return val
+    return default
```

Comment:
This doesn't do what the docstring says. Recommend change to
```python
return default if val is None or cond else val
```
though you may instead want (below is different, not a refactor)
```python
return val if val is not None or cond else default
```

Reply:
none defaults to False, so you may also want to just do cond=False

Reply:
the docstring was wrong, the code was working as intended. Docstring has been updated.

Reply:
It's still confusing. The updated docstring is still wrong in the case cond is False and val is not None. This does the docstring:
```python
def setdefault(val, default, cond=False):
    """Return val if it isn't None (or if cond is True), otherwise default."""
    return val if val is not None or cond else default
```

Reply:
The code does what it should be doing, I've updated the docstring again to make it more clear. Its a basic switch, with the default condition being `(val is not None)`, the most common use case being to assign default values when the input is `None`

Reply:
Ok cool. Now the code and docstring agree and implement what you want:
```python
def setdefault(val, default, cond=None):
    return val if cond or (cond is None and val is not None) else default
```